### PR TITLE
Summer dependency bumps

### PR DESCRIPTION
**Pull request by LebedevRI**
macOS 13.5 + Xcode 15.2, and LLVM16

@zisoft @TurboGit as per https://github.com/darktable-org/rawspeed/pull/663,
i believe this is now fine, and if no, stamp this please? :)

I'd like to get this merged to know that
https://build.opensuse.org/project/monitor/graphics:darktable:master
stays green before going any further.

Thanks!

### CODE REVIEW

**Comment by LebedevRI** [comment]
Huh, why are you running on native github images.

**Review by zisoft — APPROVED** [review]
LGTM

**Comment by LebedevRI** [comment]
@zisoft thank you!
@TurboGit ?

**Comment by victoryforce** [comment]
@LebedevRI We must support the most recent Ubuntu LTS.

**Comment by LebedevRI** [comment]
> @LebedevRI We must support the most recent Ubuntu LTS.

The question being?

**Comment by victoryforce** [comment]
> > @LebedevRI We must support the most recent Ubuntu LTS.
> 
> The question being?

This is not the question. This means we cannot require LLVM >= 16 on Linux. Please lower this requirement to LLVM 15+.

**Comment by LebedevRI** [comment]
Hm yeah, i don't quite care about clang-15 vs clang-16 at the moment, since rawspeed on linux is stuck to clang-15.
That being said, does anyone actually care about being able to build with non gcc on old ubuntu?

**Comment by LebedevRI** [comment]
Also, why does CI test GCC13 if the minimum is GCC12?

**Comment by victoryforce** [comment]
> Hm yeah, i don't quite care about clang-15 vs clang-16 at the moment, since rawspeed on linux is stuck to clang-15.

But we do care because it is our _agreed policy_ to always support the latest Ubuntu LTS release until a new one is published. Moreover, the wording "support debian stable and the most recent Ubuntu LTS", part of which I quoted, was suggested by you.

> does anyone actually care about being able to build with non gcc

Yes, as long as we declare that we support clang.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -288,8 +334,8 @@ jobs:
       fail-fast: true
       matrix:
         build:
-          - { os: macos-12,    xcode: 14.2,   deployment: 12.5 } # LLVM14, x86_64
-          - { os: macos-14,    xcode: 15.2,   deployment: 14.0 } # LLVM16, arm64
+          - { os: macos-13,    xcode: 15.2,   deployment: 13.5 } # LLVM16, x86_64
+          - { os: macos-14,    xcode: 15.2,   deployment: 13.5 } # LLVM16, arm64
```
Comment (zisoft) [inline .github/workflows/ci.yml:338]: Deployment 13.5 for macos-14 will result in lots of linker warnings. We should keep deployment 14 here.

**Comment by LebedevRI** [comment]
> > Hm yeah, i don't quite care about clang-15 vs clang-16 at the moment, since rawspeed on linux is stuck to clang-15.
> 
> But we do care

Err, i don't care in the sense that i can live with minimal version being 15 instead of 16 (:

**Review by zisoft — CHANGES_REQUESTED** [review]

**Review by zisoft — APPROVED** [review]

**Comment by victoryforce** [comment]
> Also, why does CI test GCC13 if the minimum is GCC12?

That's a good question. 🙂 The answer is that we use GCC 12 in nightly builds (there are good reasons why we can't use GCC 13 there). Therefore, any problems, if the new code is incompatible with GCC 12 (which is quite unlikely), will be detected quite quickly. Instead, CI uses GCC 13 to test the build against the latest available version of GCC.

**Comment by victoryforce** [comment]
@LebedevRI  Since we agreed to support LLVM 15 on Linux, there is no need to complicate CI so much when we can just leave CI running on the ubuntu-latest runner. I would ask you to remove this complication, there is no need for it. Thank you!

**Comment by LebedevRI** [comment]
@victoryforce thank you for taking a look!


> @LebedevRI Since we agreed to support LLVM 15 on Linux, there is no need to complicate CI so much when we can just leave CI running on the ubuntu-latest runner. I would ask you to remove this complication, there is no need for it. Thank you!

I'm not sure that makes sense.

Default github-provided image should not be relied upon,
it is always best to start with a clean slate,
because then you actually know what is going on, explicitly,
instead of happenstance.

You then also do not have to do whatever this is, which looks anything but simpler
https://github.com/darktable-org/darktable/blob/ab0e10a8445111aaf250e4bf595a004274c17f33/.github/workflows/ci.yml#L82-L91

And even if ubuntu-latest works for now, how long will it take for it to be updated to 24.04?
In the sense, will there be no LLVM compile job until that happens?
Or will we be stuck to LLVM15 until that happens? 
It just makes no sense to needlessly rely on github here.

I see that obs build-driven CI was disabled, if the desire is to be sure that things can be
built on $DISTRO, let's reenable it, and get it into working state.

**Comment by LebedevRI** [comment]
Thanks! As i've said, i'm mainly interested in seeing what shakes loose
in a way that can't be just fixed forward. Hopefully nothing much...

**Comment by LebedevRI** [comment]
https://build.opensuse.org/package/show/graphics:darktable:master/darktable looks rather green.
If there's external third-party breakage we'll hear about it eventually i guess,
but i consider the test a success, and don't expect to roll it back.

Thank you all!

**Comment by LebedevRI** [comment]
> So if Ubuntu 22.04 (which is the LTS) has LLVM 15 by default that seems a good reason for us to support it.

No ubuntu (or debian, for that matter), have clang as default compiler.

> Moving now to LLVM 16 means that next major release 4.8 on June will require this version too.

FWIW, that is not even a question. LLVM16 will be required for dt4.8,
the question is if that happens now, or in two months from now,
and i'm trying to understand for which particular reasons it can't be now.

**Comment by LebedevRI** [comment]
Ok, here's a middle ground that suits no one: how about bumping the version check,
and seeing if anyone complains within 7 days after the commit?
If there's some external CI, that will be plenty of time for it to be hit,
and if someone does manual builds, well, 7 days is pretty long.

NOTE: again, gcc is the default compiler there, if they are using clang, they are doing so manually and explicitly.

**Comment by kmilos** [comment]
[There is/was a bug in GCC 12/13 that is preventing building dt for aarch64](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1053405) (unless OpenMP is switched off), so Debian 12 and Ubuntu 22.04 are impacted. Not sure when they'll get patched...

**Comment by LebedevRI** [comment]
Yup, seen that when looking at weird build failures on https://build.opensuse.org/project/monitor/graphics:darktable:master. What a fun bug.

**Comment by LebedevRI** [comment]
> Ok, here's a middle ground that suits no one: how about bumping the version check, and seeing if anyone complains within 7 days after the commit? If there's some external CI, that will be plenty of time for it to be hit, and if someone does manual builds, well, 7 days is pretty long.
> 
> NOTE: again, gcc is the default compiler there, if they are using clang, they are doing so manually and explicitly.

... which would look something like this: https://github.com/darktable-org/darktable/pull/16374
